examples/refrigerator-app/silabs/src/RefrigeratorManager.cpp

```diff
@@ -0,0 +1,207 @@
+namespace RefAlarmAttr = chip::app::Clusters::RefrigeratorAlarm::Attributes;
+namespace TempCtrlAttr = chip::app::Clusters::TemperatureControl::Attributes;
+
+// set Parent Endpoint and Composition Type for an Endpoint
+EndpointId kRefEndpointId           = 1;
+EndpointId kColdCabinetEndpointId   = 2;
+EndpointId kFreezeCabinetEndpointId = 3;
+
+RefrigeratorManager RefrigeratorManager::sRefrigeratorMgr;
+
+namespace {
+    app::Clusters::TemperatureControl::AppSupportedTemperatureLevelsDelegate sAppSupportedTemperatureLevelsDelegate;
+
+    // Please refer to https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/namespaces
+    constexpr const uint8_t kNamespaceRefrigerator = 0x41;
+    // Refrigerator Namespace: 0x41, tag 0x00 (Refrigerator)
+    constexpr const uint8_t kTagRefrigerator = 0x00;
+    // Refrigerator Namespace: 0x41, tag 0x01 (Freezer)
+    constexpr const uint8_t kTagFreezer                                                = 0x01;
+    const Clusters::Descriptor::Structs::SemanticTagStruct::Type refrigeratorTagList[] = { { .namespaceID = kNamespaceRefrigerator,
+                                                                                            .tag         = kTagRefrigerator } };
+    const Clusters::Descriptor::Structs::SemanticTagStruct::Type freezerTagList[]      = { { .namespaceID = kNamespaceRefrigerator,
+                                                                                            .tag         = kTagFreezer } };
+} // namespace
+
+CHIP_ERROR RefrigeratorManager::Init()
+{
+    SetTreeCompositionForEndpoint(kRefEndpointId);
+    SetParentEndpointForEndpoint(kColdCabinetEndpointId, kRefEndpointId);
+    SetParentEndpointForEndpoint(kFreezeCabinetEndpointId, kRefEndpointId);
+
+    // set TagList
+    SetTagList(kColdCabinetEndpointId, Span<const Clusters::Descriptor::Structs::SemanticTagStruct::Type>(refrigeratorTagList));
+    SetTagList(kFreezeCabinetEndpointId, Span<const Clusters::Descriptor::Structs::SemanticTagStruct::Type>(freezerTagList));
+
+    app::Clusters::TemperatureControl::SetInstance(&sAppSupportedTemperatureLevelsDelegate);
+    return CHIP_NO_ERROR;
+}
+
+int8_t RefrigeratorManager::ConvertToPrintableTemp(int16_t temperature)
+{
+    constexpr uint8_t kRoundUpValue = 50;
+
+    // Round up the temperature as we won't print decimals on LCD
+    // Is it a negative temperature
+    if (temperature < 0)
+    {
+        temperature -= kRoundUpValue;
+    }
+    else
+    {
+        temperature += kRoundUpValue;
+    }
+
+    return static_cast<int8_t>(temperature / 100);
+}
+
+
+void RefrigeratorManager::RefAndTempCtrlAttributeChangeHandler(EndpointId endpointId, AttributeId attributeId, uint8_t * value, uint16_t size)
+{
+    switch (attributeId)
+    {
+        case RefAndTempAttr::CurrentMode::Id: {
+            int8_t currentMode = static_cast<int16_t>(*value);
```

Comment:
Why casting it to a 16-bit signed integer here?

Reply:
updated